snap/local/hooks/cmd/install/install.go

```diff
@@ -0,0 +1,394 @@
+	"core-command",
+	"core-data",
+	"core-metadata",
+	"support-notifications",
+	"support-scheduler",
+	"sys-mgmt-agent",
+	"device-virtual",
+	"app-service-configurable",
+}
+
+var extraConfToEnv = map[string]string{
+ 
+	"add-known-secrets": "ADD_KNOWN_SECRETS",
+	"add-registry-acl-roles": "ADD_REGISTRY_ACL_ROLES",
+}
+
+
+
+// installConfFiles copies service configuration.toml files from $SNAP to $SNAP_DATA
+func installConfFiles() error {
+	var err error
+
+	// TODO: should we continue to do this, since config overrides are
+	// the preferred way to make configuration changes now?
+	for _, v := range services {
+		destDir := hooks.SnapDataConf + "/"
+		srcDir := hooks.SnapConf + "/"
+
+		// handle exceptions (i.e. config in non-std dirs)
+		if v == "security-bootstrap-redis" {
+			destDir = destDir + "security-bootstrapper/res-bootstrap-redis"
+			srcDir = srcDir + "security-bootstrapper/res-bootstrap-redis"
+		} else if v == "app-service-configurable" {
+			destDir = destDir + v + "/res/rules-engine"
+			srcDir = srcDir + "/res/rules-engine"
+		} else {
+			destDir = destDir + v + "/res"
+			srcDir = srcDir + v + "/res"
+		}
+
+		if err = os.MkdirAll(destDir, 0755); err != nil {
+			return err
+		}
+
+		srcPath := srcDir + "/configuration.toml"
+		destPath := destDir + "/configuration.toml"
+
+		err = hooks.CopyFile(srcPath, destPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// installDevices installs device-virtual's static device files to $SNAP_DATA
+func installDevices() error {
+	var err error
+
+	destDir := hooks.SnapDataConf + "/device-virtual/res/devices"
+	destPath := destDir + "/devices.toml"
+	srcPath := hooks.SnapConf + "/device-virtual/res/devices/devices.toml"
+
+	if err = os.MkdirAll(destDir, 0755); err != nil {
+		return err
+	}
+
+	if err = hooks.CopyFile(srcPath, destPath); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// installDevProfiles installs device-virtual's device profiles to $SNAP_DATA
+func installDevProfiles() error {
+	var err error
+	profiles := []string{"bool", "float", "int", "uint", "binary"}
+
+	srcDir := hooks.SnapConf + "/device-virtual/res/profiles"
+	destDir := hooks.SnapDataConf + "/device-virtual/res/profiles"
+	if err = os.MkdirAll(destDir, 0755); err != nil {
+		return err
+	}
+
+	for _, v := range profiles {
+		fileName := "/device.virtual." + v + ".yaml"
+		srcPath := srcDir + fileName
+		destPath := destDir + fileName
+
+		if err = hooks.CopyFile(srcPath, destPath); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// installKuiper execs a shell script to install Kuiper's file into $SNAP_DATA
+func installKuiper() error {
+	setupScriptPath, err := exec.LookPath("install-setup-kuiper.sh")
+	if err != nil {
+		return err
+	}
+
+	cmdSetupKuiper := exec.Cmd{
+		Path:   setupScriptPath,
+		Args:   []string{setupScriptPath},
+		Stdout: os.Stdout,
+		Stderr: os.Stdout,
+	}
+
+	err = cmdSetupKuiper.Run()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}consul
```

Comment:
This looks like a cut & paste issue that prevents the snap from building...

Reply:
fixed